tests/ui/cbmc_checks/float-overflow/check_message_overflow.rs

```diff
@@ -0,0 +1,23 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+//
+// Check we don't print temporary variables as part of CBMC messages.
+extern crate kani;
+
+use kani::any;
+
+// Use the result so rustc doesn't optimize them away.
+fn dummy(result: f32) -> f32 {
+    result
+}
+
+#[kani::proof]
+fn main() {
+    let a = kani::any_where(|&x: &f32| x.is_finite() && x.abs() < 1e20);
```

Comment:
Is the `x.abs() < 1e20` needed?

Reply:
Also, I think it should be sufficient to just assume non NaN, i.e.
```suggestion
    let a = kani::any_where(|&x: &f32| !x.is_nan());
```
Can you check if this works?

Reply:
I added the `x.abs() < 1e20` assuming it may cause problems with multiplication. But that wasn't the case, and `let a = kani::any_where(|&x: &f32| x.is_finite());` works just fine.

 `let a = kani::any_where(|&x: &f32| !x.is_nan());` on the other hand is failing since the individual numbers themselves are not NaN but the result is overflowing. --> Getting dereferencing errors.

Reply:
> on the other hand is failing since the individual numbers themselves are not NaN but the result is overflowing. --> Getting dereferencing errors.

Can you post the failure? Since we're disabling CBMC's float overflow checks, I wouldn't expect it to fail.

Reply:
For example, this program runs without issues: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c8191caef9b78b217aa6654a01483f20

Reply:
Oh, but perhaps multiplying `f32::INFINITY` by 0.0 results in a NaN and CBMC flags this because of the NaN checks?

Reply:
Yes. It is being multiplied by 0, and also due to `f32::INFINITY` - `f32::INFINITY`

Reply:
Confirmed offline that CBMC's NaN checks fail if we only restrict the values to `!is_nan()`. Perhaps we should also consider disabling CBMC's NaN checks since produce a NaN does not seem to be UB.

Reply:
Can you file a tracking issue to consider disabling CBMC's NaN checks as well?

Reply:
Sure. #3875